### PR TITLE
Add docs on custom image

### PR DESCRIPTION
**Pull request by anyasabo**
We suggest creating custom images but there weren't clear docs from us or ES on how exactly to do that. This adds an example. It may be worthwhile to improve the ES docs as well as the customization there is pretty basic.

### CODE REVIEW

**Comment by jordansissel** [comment]
Content looks good!

File: docs/images.asciidoc [hunk 1]
```diff
@@ -0,0 +1,42 @@
+[id="{p}-custom-images"]
+== Creating custom images
+
+It is possible to create your own custom Elasticsearch (or Kibana, etc.) image instead of using the base image provided by Elastic. You might want to do this to preload plugins in the image rather than having to install them via init container each time a pod starts. For example, if you want to create an Elasticsearch 7.3.2 image with the https://www.elastic.co/guide/en/elasticsearch/plugins/master/repository-gcs.html[Google Cloud Storage Repository Plugin], you could create a `Dockerfile` containing
+
+----
+FROM docker.elastic.co/elasticsearch/elasticsearch:7.3.2
```
Comment (pebrc) [inline docs/images.asciidoc]: Should we mention that you have to use the official image as the base image for things to work out?

File: docs/images.asciidoc [hunk 2]
```diff
@@ -0,0 +1,42 @@
+[id="{p}-custom-images"]
+== Creating custom images
+
+It is possible to create your own custom Elasticsearch (or Kibana, etc.) image instead of using the base image provided by Elastic. You might want to do this to preload plugins in the image rather than having to install them via init container each time a pod starts. For example, if you want to create an Elasticsearch 7.3.2 image with the https://www.elastic.co/guide/en/elasticsearch/plugins/master/repository-gcs.html[Google Cloud Storage Repository Plugin], you could create a `Dockerfile` containing
```
Comment (thbkrkr) [inline docs/images.asciidoc]: Maybe add a link to https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-init-containers-plugin-downloads.html on `init container`.

File: docs/images.asciidoc [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+----
+FROM docker.elastic.co/elasticsearch/elasticsearch:7.3.2
+RUN bin/elasticsearch-plugin install --batch repository-gcs
+----
+
+Build the image with
```
Comment (thbkrkr) [inline docs/images.asciidoc]: Finish with a colon?

File: docs/images.asciidoc [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+----
+docker build --tag elasticsearch-gcs:7.3.2 .
+----
+
+There are various hosting options for your images. If you use Google Kubernetes Engine, it is automatically configured to use the Google Container Registry (see https://cloud.google.com/container-registry/docs/using-with-google-cloud-platform#google-kubernetes-engine[here] for more information). To use the image, you could then https://cloud.google.com/container-registry/docs/pushing-and-pulling#pushing_an_image_to_a_registry[push to the registry] with
```
Comment (thbkrkr) [inline docs/images.asciidoc]: Finish with a colon?

File: docs/images.asciidoc (hunk 2 above)
Comment (thbkrkr) [inline docs/images.asciidoc]: Finish with a colon?

File: docs/snapshots.asciidoc [hunk 5]
```diff
@@ -16,7 +16,7 @@ For more information on Elasticsearch snapshots, see https://www.elastic.co/guid
 [id="{p}-install-plugin"]
 ==== Install the storage repository plugin
 
-To install the storage repository plugin, you can either use a custom image or add your own init container which
+To install the storage repository plugin, you can either use a link:k8s-images.html[custom image] or add your own init container which
```
Comment (thbkrkr) [inline docs/snapshots.asciidoc]: Same here, add a link to the k8s-init-containers-plugin-downloads.html doc page?

**Pull request merged**